src/dependency-manager/manager.ts

```diff
@@ -371,6 +370,20 @@ export default abstract class DependencyManager {
   }
 
   validateGraph(graph: DependencyGraph): void {
+    // Check for protocols that are not http or https
+    for (const node of graph.nodes) {
+      if (node.is_external) continue;
+      if (!(node instanceof ComponentNode)) continue;
+
+      const component_interfaces = node.config.interfaces;
+      for (const [component_name, component_interface] of Object.entries(component_interfaces)) {
+        const valid_protocol = [undefined, 'http', 'https'];
```

Comment:
Move this out since it is a static definition. Could be a private readonly of the class.
Also make plural.

Reply:
Sounds good!